cocos/audio/android/AudioDecoder.cpp

```diff
@@ -468,8 +477,11 @@ bool AudioDecoder::decodeToPcm()
 
     ALOGV("Stopped decoding");
 
-    /* Destroy the UrlAudioPlayer object */
-    SL_DESTROY_OBJ(_playObj);
+    /* Destroy the AudioPlayer object */
+    {
+        std::lock_guard<std::mutex> lk(__SLPlayerMutex);
+        SL_DESTROY_OBJ(_playObj);
+    }
```

Comment:
It's better to add the braces since `std::lock_guard` needs it to unlock mutex right after SL_DESTROY_OBJ. Otherwise, the `unlock` operation will be delayed to the function return.